application/src/main/java/org/opentripplanner/transit/api/request/TripOnServiceDateRequest.java

```diff
@@ -1,77 +1,77 @@
 package org.opentripplanner.transit.api.request;
 
 import java.time.LocalDate;
-import java.util.List;
+import org.opentripplanner.transit.api.model.FilterValueCollection;
 import org.opentripplanner.transit.model.framework.FeedScopedId;
 import org.opentripplanner.transit.model.timetable.TripAlteration;
-import org.opentripplanner.utils.collection.ListUtils;
+import org.opentripplanner.transit.model.timetable.TripOnServiceDate;
 
-/*
+/**
  * A request for trips on a specific service date.
- *
- * This request is used to retrieve TripsOnServiceDates that match the provided criteria.
+ * </p>
+ * This request is used to retrieve {@link TripOnServiceDate}s that match the provided filter values.
  * At least one operatingDay must be provided.
  */
 public class TripOnServiceDateRequest {
 
-  private final List<LocalDate> operatingDays;
-  private final List<FeedScopedId> authorities;
-  private final List<FeedScopedId> lines;
-  private final List<FeedScopedId> serviceJourneys;
-  private final List<FeedScopedId> replacementFor;
-  private final List<String> privateCodes;
-  private final List<TripAlteration> alterations;
+  private final FilterValueCollection<LocalDate> serviceDates;
+  private final FilterValueCollection<FeedScopedId> agencies;
+  private final FilterValueCollection<FeedScopedId> routes;
+  private final FilterValueCollection<FeedScopedId> serviceJourneys;
+  private final FilterValueCollection<FeedScopedId> replacementFor;
+  private final FilterValueCollection<String> netexInternalPlanningCodes;
+  private final FilterValueCollection<TripAlteration> alterations;
 
   protected TripOnServiceDateRequest(
-    List<LocalDate> operatingDays,
-    List<FeedScopedId> authorities,
-    List<FeedScopedId> lines,
-    List<FeedScopedId> serviceJourneys,
-    List<FeedScopedId> replacementFor,
-    List<String> privateCodes,
-    List<TripAlteration> alterations
+    FilterValueCollection<LocalDate> serviceDates,
+    FilterValueCollection<FeedScopedId> agencies,
+    FilterValueCollection<FeedScopedId> routes,
+    FilterValueCollection<FeedScopedId> serviceJourneys,
+    FilterValueCollection<FeedScopedId> replacementFor,
+    FilterValueCollection<String> netexInternalPlanningCodes,
+    FilterValueCollection<TripAlteration> alterations
   ) {
-    if (operatingDays == null || operatingDays.isEmpty()) {
+    if (serviceDates == null || serviceDates.get() == null || serviceDates.get().isEmpty()) {
```

Comment:
`serviceDates` should never be null here as far as I can see, so could remove the first null check.

Also the other checks will never return true since they would cause the FilterValueCollection to throw an exception. It looks like serviceDates is required to be a non-null non-empty list, so maybe it shouldn't be a FilterValueCollection? Or at least it should probably not be constructed by the `ofEmptyIsEverything()` factory method since empty does not mean allow everything for serviceDates. Perhaps there should be a `ofEmptyIsInvalid()` factory method for this case?

Reply:
Good call. Implementing the ofEmptyIsInvalid factory method and creating the appropriate subclasses of the FilterValues class.